### DISCUSSION TITLE
github actions & jenkins builds

### DISCUSSION THREAD

**Discussion by SteveSchafer-Innovent** [eclipse-birt/birt#1806] [Q&A]
Just curious why the BIRT project is built twice:  In githib/actions and also in ci.eclipse.org/birt.  Will one of those be going away at some point?

**Comment by merks** [comment]
It’s verified/built twice for PR builds. Only ci.eclipse.org produces builds that are signed and published and only for the master branch.  There are no plans to change that though the GitHub action is redundant.  Perhaps we should consider the planet…

**Comment by SteveSchafer-Innovent** [comment]
I want to revisit issue 625 (Publish maven build to central).  I worked on that a couple of years ago but wasn't successful and had to drop it.  Now I think I have better ideas for how to accomplish it.  To that end I'd like to create a branch to play with it.  Here are some follow-on questions:

1. In ci.eclipse.org, is the build controlled by Jenkinsfile?  If I want to introduce scripts, would Jenkinsfile be the place to do it?
2. Back a couple of years ago I had a branch but it looks like it got deleted.  I was experimenting with the jenkins builds but I can't find those old build logs and configurations.  Would they be backed up somewhere or are they gone for good?
3. In those old builds I was able to introduce scripts in the Jenkins configuration, but the configuration looks different now.  Did the configuration change or am I just not looking in the right place?

**Reply by merks** [reply]
Did you read the PDF here?

https://github.com/eclipse-birt/birt/issues/625#issuecomment-2238695528

Both the Platform and EMF publish based on the contents of their p2 repository.

A "dump" of the poms used for the Tycho build is not going to produce quality assets.  What I see here  also does not look promising:

https://repo1.maven.org/maven2/com/innoventsolutions/birt/runtime/

---

I don't know if deleted jobs are backed up somewhere.  I don't think so but you could ask here:

https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/

Yes, the https://ci.eclipse.org/birt/job/build/ job is defined by https://github.com/eclipse-birt/birt/blob/master/Jenkinsfile

You can create other types of job on the ci instance, not just multi-branch pipeline jobs:

![image](https://github.com/user-attachments/assets/c4f955e1-5122-45b0-aef2-586a76be2d4d)
